templates/streamhost/streamhost.erb

```diff
@@ -18,6 +18,9 @@ server {
   listen [<%= @ipv6_listen_ip %>]:<%= @ipv6_listen_port %> <% if @ipv6_listen_options %><%= @ipv6_listen_options %><% end %>;
   <%- end -%>
 <%- end -%>
+<%- if Array(@resolver).count > 0 -%>
```

Comment:
The puppet code already enforces that resolver is an array.  No need to cast it.

```suggestion
<%- unless @resolver.empty -%>
```

Reply:
(There's no type checking on raw_prepend/raw_append and the docs say it takes strings or arrays of strings, so the casting is needed on those lines)